flask/app/variants.py

```diff
@@ -358,6 +471,21 @@ def summary(type: str):
             abort(500, "Unexpected error")
 
         agg_df = get_report_df(sql_df, type=type)
+
+        columns = request.args.get("columns", type=str)
+        if columns is not None:
+            columns = columns.split(",")
+            # filter out faulty columns
+            valid_columns = {col for col in agg_df.columns.values}
+            fixed_columns = [col for col in columns if col in valid_columns]
+            if len(fixed_columns) < len(columns):
+                app.logger.warn(
+                    "Ignoring invalid columns from request: {}".format(
+                        set(columns) ^ set(fixed_columns)
+                    )
+                )
+            agg_df = agg_df.loc[:, fixed_columns]
```

Comment:
This looks good for the csv, but there's also the question of the JSON representation (which I don't think is being used anywhere at the moment). Is it possible to move the filter into the SQL/ORM as a select statement so that only the requested columns are returned from the database? I'm not sure how this would impact current functionality, esp. re: Pandas, but in theory it might take care of the csv and JSON responses in one go.

Reply:
Interested to see how this goes as I believe duplicate columns end up being selected as a result from the eager loading

Reply:
The way the JSON result is formatted for the participant-wise report also adds to the complexity. Currently hoping to find a way to get a set of columns that are present in the query, and then only accept columns that are a subset of those columns, similar to the CSV case.

Reply:
Not sure how to proceed with the column selection issue. Currently, I have a way to generate a set of model entities using the requested list of column names, however I've tried a couple of different approaches for reducing the query down to the requested columns:

- `query.options(load_only(...))` allows you to only load a specific subset of columns/entities, and will "defer" loading for any other entities. This doesn't really address the issue, since it effectively just delays loading other columns which we don't want to load to begin with.
- `db.session.query(...)` is how you specify what to SELECT. However, this seems to cause issues with relationship properties (in particular, `Variant.genotype`). This is the approach that is most similar to what Conor suggested, but I haven't yet found a way to make it work due to this relationship property issue.
- `query.with_entities(...)` allows you to return a subset of the columns selected as a list of tuples in the same order as they're provided to the function. This approach seems the most like what we're looking for, however it means that we can no longer use `asdict` to convert the query output, since the output is now a list of tuples and not a list of gene/variant class instances. It also doesn't seem to play nicely with pandas, but I haven't looked into that aspect yet.

Going to work on the frontend side of this issue for a bit and come back to this later.